crates/unftp-sbe-gcs/src/lib.rs

```diff
@@ -134,30 +138,102 @@ impl CloudStorage {
     {
         let client: Client<HttpsConnector<HttpConnector<GaiResolver>>, Body> =
             Client::builder().build(HttpsConnectorBuilder::new().with_native_roots().https_or_http().enable_http1().build());
-        CloudStorage {
+        Self {
             client,
             auth: auth.into(),
             uris: GcsUri::new(base_url.into(), bucket.into(), root),
+
+            cached_token: Default::default(),
         }
     }
 
-    // TODO: Cache the token. For `ServiceAccountKey`, the oauth client would already cache the token - we just need to move it to `CloudStorage`. For `WorkloadIdentity`, we can cache it in `CloudStorage`.
     #[tracing_attributes::instrument]
-    async fn get_token(&self) -> Result<String, Error> {
+    async fn get_token_value(&self) -> Result<String, Error> {
```

Comment:
maybe a bit pedantic but why does this need to be renamed. It feels like its mostly noise and not a better name

Reply:
This function does not return the whole token, but just the value. A complete OAuth2 access token consists of the token value, type, expiry information, refresh token, and scope.

Reply:
I'm happy with the rename, its more correct and was done because there is also now a fetch_token that returns the full token.